lib/log-events-processor.js

```diff
@@ -0,0 +1,94 @@
+import Redis from 'ioredis'
+import {
+  cloudwatchPutLogEvents,
+  cloudwatchDescribeLogStreams,
+} from './cloudwatch-logs'
+
+const redis = new Redis()
+
+export const tokenKey = (logGroupName, logStreamName) => (
+  `${logGroupName}-${logStreamName}-next-token`
+)
+
+export const getValidSequenceToken = (logGroupName, logStreamName) => (
+  redis.get(tokenKey(logGroupName, logStreamName))
+)
+
+export const saveValidSequenceToken = (token, logGroupName, logStreamName) => (
+  redis.set(tokenKey(logGroupName, logStreamName), token)
+)
+
+export const putLogEvents = job => token => {
+  const { data: { logGroupName, logStreamName, logEvents } } = job
+
+  const params = {
+    sequenceToken: token,
+    logGroupName,
+    logStreamName,
+    logEvents,
+  }
+
+  return cloudwatchPutLogEvents(params)
+}
+
+export const completeJobSuccessfully = (job, done) => cloudwatchResponse => {
+  const { data: { logGroupName, logStreamName } } = job
+
+  return saveValidSequenceToken(
+    cloudwatchResponse.nextSequenceToken,
+    logGroupName,
+    logStreamName,
+  ).then(() => { done(null, cloudwatchResponse) })
+}
+
+export const shouldRefetchValidToken = errorCode => (
+  errorCode === 'InvalidSequenceTokenException' ||
+  errorCode === 'DataAlreadyAcceptedException'
```

Comment:
J'ajouterais un peu d'indentation pour faire comprendre le lien avec la ligne précédente ici.

```suggestion
    errorCode === 'DataAlreadyAcceptedException'
```